src/modeler/loadFiles/readLibraries.cpp

```diff
@@ -85,9 +105,103 @@ std::vector<ModelerStudy::SystemModel::Library> loadLibraries(const fs::path& st
+    {
+        throw std::runtime_error("No library '" + libId + "' found.");
+    }
+
+    auto modelIt = std::ranges::find_if(lib->models,
+                                        [&modelId](const auto& m) { return m.id == modelId; });
+    if (modelIt == lib->models.end())
+    {
+        throw std::runtime_error("No model '" + modelId + "' found.");
+    }
+
+    return *modelIt;
+}
+
+void updateVariables(YmlModel::Model& model, const YmlOptimConfig::Model& optimConfigModel)
+{
+    for (const auto& optCfgVar: optimConfigModel.variables)
+    {
+        auto predicate = [&optCfgVar](const auto& v) { return v.id == optCfgVar.id; };
+        auto var = std::ranges::find_if(model.variables, predicate);
+        if (var == model.variables.end())
+        {
+            throw std::runtime_error("No variable '" + optCfgVar.id + "' found.");
+        }
+        var->location = optCfgVar.location;
+    }
+}
+
+void updateConstraints(YmlModel::Model& model, const YmlOptimConfig::Model& optimConfigModel)
+{
+    for (const auto& optCfgConst: optimConfigModel.constraints)
+    {
+        auto predicate = [&optCfgConst](const auto& c) { return c.id == optCfgConst.id; };
+        auto constraint = std::ranges::find_if(model.constraints, predicate);
+        if (constraint == model.constraints.end())
+        {
+            throw std::runtime_error("No constraint '" + optCfgConst.id + "' found.");
+        }
+        constraint->location = optCfgConst.location;
+    }
+}
+
+void updateObjective(YmlModel::Model& model, const YmlOptimConfig::Model& optimConfigModel)
+{
+    for (const auto& optCfgObj: optimConfigModel.objectives)
+    {
+        auto predicate = [&optCfgObj](const auto& obj) { return obj.id == optCfgObj.id; };
+        auto objective = std::ranges::find_if(model.objectives, predicate);
+        if (objective == model.objectives.end())
+        {
+            throw std::runtime_error("No objective '" + optCfgObj.id + "' found.");
+        }
+        objective->location = optCfgObj.location;
+    }
+}
+
+void updateSystemModel(YmlModel::Model& model, const YmlOptimConfig::Model& optimConfigModel)
+{
+    updateVariables(model, optimConfigModel);
+    updateConstraints(model, optimConfigModel);
+    updateObjective(model, optimConfigModel);
+}
+
+void updateLibrariesWithOptimConfig(std::vector<YmlModel::Library>& ymlLibs,
+                                    const YmlOptimConfig::OptimConfig& ymlOptimConfig)
+{
+    for (const auto& optimConfigModel: ymlOptimConfig)
+    {
+        auto& model = fetchModelInLibrairies(optimConfigModel, ymlLibs);
+        updateSystemModel(model, optimConfigModel);
+    }
+}
+
+std::vector<SystemModel::Library> loadLibraries(const fs::path& studyPath)
+{
+    auto ymlLibraries = loadLibrariesFromYaml(studyPath);
+    auto ymlOptimConfig = loadOptimConfigFromYaml(studyPath);
+    updateLibrariesWithOptimConfig(ymlLibraries, ymlOptimConfig);
+    return convertIntoSystemLibs(ymlLibraries);
```

Comment:
Very clear ! I think `ymlOptimConfig` won't be modified, so it can be marked as `const`.
```suggestion
    auto ymlLibraries = loadLibrariesFromYaml(studyPath);
    const auto ymlOptimConfig = loadOptimConfigFromYaml(studyPath);
    updateLibrariesWithOptimConfig(ymlLibraries, ymlOptimConfig);
    return convertIntoSystemLibs(ymlLibraries);
```

Reply:
Done